tests/performance/joins_in_memory_pmj.xml

```diff
@@ -38,5 +38,15 @@
     <query tag='LEFT ON (noopt)'>SELECT COUNT() FROM ints l LEFT JOIN ints r ON l.i64 = r.i64 WHERE i32 = 200042 SETTINGS partial_merge_join_optimizations = 0</query>
     <query tag='LEFT IN (noopt)'>SELECT COUNT() FROM ints l LEFT JOIN ints r USING i64 WHERE i32 IN(42, 100042, 200042, 300042, 400042) SETTINGS partial_merge_join_optimizations = 0</query>
     
+    <query tag='RIGHT'>SELECT COUNT() FROM ints l RIGHT JOIN ints r USING i64 WHERE i32 = 200042</query>
```

Comment:
There is no 200042 in those tables, remember? I'm fixing it here https://github.com/ClickHouse/ClickHouse/pull/11886, should be ready by the end of the week.

Some tests are very bad even with the fixes, though:
https://clickhouse-test-reports.s3.yandex.net/11886/10b6f98ca6de50de0f32d95b5e65b50d3b27fef3/performance_comparison/report.html#5.45

The red 'p<0.001 threshold' number means that we expect to see variability of 60% even when comparing the same version to itself. Is there some underlying process that can significantly change the query time, like, something is synched to disk?

Reply:
> There is no 200042 in those tables, remember?

I thought it had been allready fixed. I can remove one 0 from all join_in_memory tests in this PR if needed. Should I?

>  Is there some underlying process that can significantly change the query time, like, something is synched to disk?

No the tests are expected to be in-memory so there should be no disk or other long activity. But there could be lots of memory allocations. There's also no NUMA awareness.
Do we have flaps only for MergeJoin or both hash and merge joins are affected?

Reply:
> > There is no 200042 in those tables, remember?
> 
> I thought it had been allready fixed. I can remove one 0 from all join_in_memory tests in this PR if needed. Should I?
> 

Yes, please do, so that the new queries also make sense. Better copy it from my PR because I vaguely remember I had to make some other tweaks like changing the table size.

> > Is there some underlying process that can significantly change the query time, like, something is synched to disk?
> 
> No the tests are expected to be in-memory so there should be no disk or other long activity. But there could be lots of memory allocations. There's also no NUMA awareness.
> Do we have flaps only for MergeJoin or both hash and merge joins are affected?

The queries in the table I linked are especially bad, others are 20% which is not so good but acceptable. Here is a full list:
https://clickhouse-test-reports.s3.yandex.net/11886/10b6f98ca6de50de0f32d95b5e65b50d3b27fef3/performance_comparison/all-queries.html#2.856

Looks like it's related to cache misses in radix sort. For example, here are the most unstable metrics for pmj test, query no. 6. I got them from perf test result archive `output.7z`, the file is `join_in_memory_pmj_6_SELECT.....right.metrics.rep`. The array in second column is [min, med, max] number for this metric among query runs:
```
8.059	[26625473,31851984,283323775]	PerfCacheReferences
6.25	[4000,16000,104000]	SystemTimeMicroseconds
5.25	[2,4,23]	RadixSort.h:158#void RadixSort<DB::(anonymous namespace)::RadixSortTraits<long> >::radixSortLSDInternal<true>(DB::(anonymous namespace)::ValueWithIndex<long>*, unsigned long, bool, unsigned long*)(Real)
4.333	[2,6,28]	RadixSort.h:299#void RadixSort<DB::(anonymous namespace)::RadixSortTraits<long> 
...
3.417	[105954,387394,1429720]	PerfCacheMisses
3.309	[35,42,174]	PODArray.h:268#DB::ColumnVector<long>::getPermutation(bool, unsigned long, int, DB::PODArray<unsigned long, 4096ul, Allocator<false, false>, 15ul, 16ul>&) const(Real)
...
```

Same for no. 7:
```
5.5	[1,2,12]	RadixSort.h:300#void RadixSort<DB::(anonymous namespace)::RadixSortTraits<long> >::radixSortLSDInternal<true>(DB::(anonymous namespace)::ValueWithIndex<long>*, unsigned long, bool, unsigned long*)(Real)
3.5	[1,2,8]	Set.cpp:406#void DB::Set::executeImplCase<DB::SetMethodOneNumber<unsigned int, HashSetTable<unsigned int, HashTableCell<unsigned int, HashCRC32<unsigned int>, HashTableNoState>, HashCRC32<unsigned int>, HashTableGrower<8ul>, Allocator<true, true> >, true>, false>(DB::SetMethodOneNumber<unsigned int, HashSetTable<unsigned int, HashTableCell<unsigned int, HashCRC32<unsigned int>, HashTableNoState>, HashCRC32<unsigned int>, HashTableGrower<8ul>, Allocator<true, true> >, true>&, std::__1::vector<DB::IColumn const*, std::__1::allocator<DB::IColumn const*> > const&, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul>&, bool, unsigned long, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul> const*) const(Real)
3	[1,2,7]	clickhouse#DB::ColumnVector<long>::compareAt(unsigned long, unsigned long, DB::IColumn const&, int) const(Real)
2.8	[4000,20000,60000]	SystemTimeMicroseconds
2.666	[1,3,9]	RadixSort.h:324#void RadixSort<DB::(anonymous namespace)::RadixSortTraits<long> >::radixSortLSDInternal<true>(DB::(anonymous namespace)::ValueWithIndex<long>*, unsigned long, bool, unsigned long*)(Real)
2.537	[11640,44021,123350]	PerfDataTLBMisses
2.5	[2,6,17]	PerfCpuMigrations
2.398	[251026965,360016441,1114543542]	PerfMinEnabledRunningTime
1.939	[268862,476398,1192965]	PerfCacheMisses
1.833	[388,1138,2474]	OSCPUWaitMicroseconds
1.714	[3,7,15]	RadixSort.h:299#void RadixSort<DB::(anonymous namespace)::RadixSortTraits<long> >::radixSortLSDInternal<true>(DB::(anonymous namespace)::ValueWithIndex<long>*, unsigned long, bool, unsigned long*)(Real)
```

This is not related to this PR, but we'll have to do something about these tests, having only 60% sensitivity is not so good.